### PR TITLE
Fix reports stuck in deleting

### PR DESCRIPTION
**Pull request by pathakh**

### CODE REVIEW

File: app/jobs/nfg_csv_importer/destroy_import_job.rb [hunk 1]
```diff
@@ -24,7 +26,7 @@ def perform(*args)
 
       import.reload
 
-      if import.imported_records.where(deleted: true).count == import.imported_records.count
+      if ((import.imported_records.where(deleted: true).count) == (import.imported_records.count - non_deletable_records_count))
```
Comment (hoenth) [inline app/jobs/nfg_csv_importer/destroy_import_job.rb]: There are a couple issues with this. 

The non_deleteable_records_count would only be for this batch, not all batches. So when comparing the counts, above logic would not work if there were multiple batches and each batch had some non deletable records. 

Also, doesn't the non_deletable_records_count only include those records which were originally created but could not be deleted. The other group are those records that were updated. so the above might need to be: 
```
if ((import.imported_records.where(deleted: true).count) == (import.imported_records.created.count - non_deletable_records_count))
```

File: spec/jobs/destroy_import_job_spec.rb [hunk 2]
```diff
@@ -34,14 +31,33 @@
...
 
+  describe "For the last batch that is processed" do
+    let(:number_of_imported_records) { 2 }
+    let(:batch) { imported_records.map(&:id)}
+
+    it_behaves_like "processing after the last batch"
+
+    it_behaves_like "destroying the imported record"
+
+    context 'when there is a record that cannot be destroyed' do
```
Comment (hoenth) [inline spec/jobs/destroy_import_job_spec.rb]: You'll need to add a context for when the record was updated instead of created.

**Comment by pathakh** [comment]
@hoenth  you mentioned you wanted to take another look at this.  can i merge this into uat if you are ok with it.

**Comment by hoenth** [comment]
@pathakh Go ahead and merge it. I will take a look today

File: spec/jobs/destroy_import_job_spec.rb [hunk 3]
```diff
@@ -34,14 +31,45 @@
...
-    it_behaves_like "destroying the imported record"
-
     it 'enqueues a queued status email' do
       NfgCsvImporter::ImportMailer.expects(:send_import_result).returns(mock('import_result', deliver_now: nil))
       subject
     end
   end
 
+  describe "For the last batch that is processed" do
+    let(:number_of_imported_records) { 2 }
```
Comment (hoenth) [inline spec/jobs/destroy_import_job_spec.rb]: @pathakh Where do you test that the imported_record gets changed from "create" to "non_deletable"

Comment (pathakh) [reply spec/jobs/destroy_import_job_spec.rb]: @hoenth oops, i've added this spec now.  i was able to get this spec from fail to pass with the changes.  but didn't test the specific change to non_deletable.

**Pull request merged**